libs/blocks/preflight/panels/seo.js

```diff
@@ -133,25 +135,109 @@ async function checkLorem() {
+    linksResult.value = {
+      icon: fail,
+      title: 'Links',
+      description: `A VPN connection is required to use the link check service.
+      Please turn on VPN and refresh the page. If VPN is running contact your site engineers for help.`,
+    };
+  };
+
+  // Check to see if Spidy is available.
+  try {
+    const resp = await fetch(spidy.url, { method: 'HEAD' });
+    if (!resp.ok) {
+      connectionError();
+      return;
+    }
+  } catch (e) {
+    connectionError();
+    console.error(`There was a problem connecting to the link check API ${spidy.url}. ${e}`);
+    return;
+  }
+
   const result = { ...linksResult.value };
-  const links = document.querySelectorAll('a[href^="/"]');
 
-  let badLink;
-  for (const link of links) {
-    const resp = await fetch(link.href, { method: 'HEAD' });
-    if (!resp.ok) badLink = true;
+  /* Find all links.
+   * Remove any local or existing preflight links.
+   * Set link to use hlx.live
+   * */
+  const links = [...document.querySelectorAll('a')]
+    .filter((link) => {
+      if (!link.href.includes('local') && !link.closest('.preflight')) {
+        link.dataset.liveHref = link.href.replace('hlx.page', 'hlx.live');
+        return true;
+      }
+      return false;
+    });
+  const groups = makeGroups(links);
+
+  for (const group of groups) {
+    const urls = group.map((link) => {
+      const liverHref = link.dataset.liveHref;
+      return liverHref;
+    });
+    const opts = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ urls }),
+    };
+    try {
+      const resp = await fetch(`${spidy.url}/api/url-http-status`, opts);
+      if (!resp.ok) return;
+
+      const json = await resp.json();
+      if (!json) return;
+      json.data.forEach((linkResult) => {
+        const status = linkResult.status === 'ECONNREFUSED' ? 503 : linkResult.status;
+        // Response will come back out of order, use ID to find the correct index
+        group[linkResult.id].status = status;
+
+        if (status >= 399) {
+          let parent = '';
+          if (group[linkResult.id].closest('header')) parent = 'Gnav';
+          if (group[linkResult.id].closest('main')) parent = 'Main content';
+          if (group[linkResult.id].closest('footer')) parent = 'Footer';
+          badLinks.value = [...badLinks.value,
+            {
+              // Diplay .hlx.live URL in broken link list for relative links
+              href: group[linkResult.id].dataset.liveHref,
+              status: group[linkResult.id].status,
+              parent,
+            }];
+          group[linkResult.id].classList.add('broken-link');
+          group[linkResult.id].dataset.status = status;
+        }
+      });
+    } catch (e) {
+      console.error(`There was a problem connecting to the link check API ${spidy.url}/api/url-http-status. ${e}`);
+    }
   }
 
-  if (badLink) {
+  if (badLinks.value.length > 0) {
```

Comment:
super duper nit - I've always preferred `if (badLinks.value.length) {` but doesn't really matter.

Out of curiosity I checked milo codebase:
`.length > 0)` used 81 times
`.length)` used 414 times